lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemConfig.java

```diff
@@ -266,4 +274,16 @@ public S3FileSystemConfig setHttpProxySecure(boolean httpProxySecure)
         this.httpProxySecure = httpProxySecure;
         return this;
     }
+
+    public S3ObjectStorageClassFilter getS3ObjectStorageClassFilter()
+    {
+        return s3ObjectStorageClassFilter;
+    }
+
+    @Config("s3.object-storage-class-filter")
```

Comment:
if i understand correctly, this is applicable to hive connector only and should be a config there.
Also, i don't know why anyone would want to silently skip glacier objects. that's just returning wrong data, isn't it?

delta and iceberg use listings for cleanups (eg DROP TABLE); in such case I don't think we should be skipping glacier objects.

Reply:
That's a good point, although this is actually a problem with the legacy S3 code. Thoughts on how to improve this?

Reply:
> if i understand correctly, this is applicable to hive connector only and should be a config there. delta and iceberg use listings for cleanups (eg DROP TABLE); in such case I don't think we should be skipping glacier objects.

Good point. What about overloading `listFiles()` and adding a function signature which accepts a filter (and defaults to the filesystem configuration for the regular signature of `listFiles(Location location)`? The `S3AFileSystem` does something similar.

> Also, i don't know why anyone would want to silently skip glacier objects. that's just returning wrong data, isn't it?

This is the default behavior in Athena. Lots of people have lifecycle policies on S3 buckets which transition their objects in place from standard storage to Glacier storage. This allows them to continue to read from that table.